common/src/main/kotlin/io/homeassistant/companion/android/common/notifications/PushProvider.kt

```diff
@@ -0,0 +1,91 @@
+package io.homeassistant.companion.android.common.notifications
+
+import android.content.Context
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import io.homeassistant.companion.android.common.BuildConfig
+import io.homeassistant.companion.android.common.data.integration.DeviceRegistration
+import io.homeassistant.companion.android.common.data.servers.ServerManager
+import io.homeassistant.companion.android.database.AppDatabase
+import io.homeassistant.companion.android.database.settings.PushProviderSetting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+interface PushProvider {
```

Comment:
Let's make sure this public interface is properly documented. Also it would be nice to create a section in https://developers.home-assistant.io/docs/android/ about how to add a new provider if this is meant to evolve.